src/core.c

```diff
@@ -802,6 +808,124 @@ static int find_gpu_user_mode_driver(enum EnGpuDrv gpu_driver, char *user_mode_d
+
+	for (i = 1; i <= num_pf; i++) // find GPU device
+	{
+		ret_cl = clGetPlatformIDs(i, &pf_id, NULL); // get platform id
+		if (ret_cl)
+		{
+			MSG_ERROR(_("failed to call clGetPlatformIDs"), NULL);
+			return ret_cl;
+		}
+
+		ret_cl = clGetDeviceIDs(pf_id, CL_DEVICE_TYPE_GPU, 0, NULL, &num_ocl_dev); // get number of device
+		if (ret_cl)
+		{
+			MSG_ERROR(_("failed to call clGetDeviceIDs"), NULL);
+			return ret_cl;
+		}
+		MSG_DEBUG("Number of OpenCL Device:\t%d", num_ocl_dev);
+
+		for (j = 1; j <= num_ocl_dev; j++)
+		{
+			ret_cl = clGetDeviceIDs(pf_id, CL_DEVICE_TYPE_GPU, i, &ocl_dev_id, NULL); // get deviceid of opencl
+			if (ret_cl)
+			{
+				MSG_ERROR(_("faild to call clGetDeviceIDs"), NULL);
+				return ret_cl;
+			}
+
+			CLINFO(ocl_dev_id, CL_DEVICE_VENDOR_ID, ocl_vendor);
+			if (dev->vendor_id != ocl_vendor)
+				break;
+
+			switch (ocl_vendor)
+			{
+			case 0x1002:
+				ret_cl = CLINFO(ocl_dev_id, CL_DEVICE_TOPOLOGY_AMD, topo_amd);
+				if (ret_cl)
+				{
+					MSG_WARNING(_("OpenCL Driver does not support CL_DEVICE_TOPOLOGY_AMD"), NULL);
+					return ret_cl;
+				}
+
+				if (dev->bus  ==  topo_amd.pcie.bus     &&
+				    dev->dev  ==  topo_amd.pcie.device  &&
+				    dev->func ==  topo_amd.pcie.function)
+				{
+					ret_cl = CLINFO(ocl_dev_id, CL_DEVICE_GFXIP_MAJOR_AMD, amd_gfx_major);
+					if (ret_cl)
+					{
+						MSG_WARNING(_("OpenCL Driver does not support CL_DEVICE_GFXIP_MAJOR_AMD"), NULL);
+						amd_gfx_major = 0;
+					}
+
+					if (10 <= amd_gfx_major) // for RDNA, gfx10+
+						snprintf(comp_unit_type, MAXSTR, "%s", "WGP"); // Workgroup Processor
+					else
+						snprintf(comp_unit_type, MAXSTR, "%s", "CU"); // Compute Unit
+
+					CLINFO(ocl_dev_id, CL_DEVICE_MAX_COMPUTE_UNITS, *comp_unit);
+				}
+				break;
+			case 0x8086:
+				CLINFO(ocl_dev_id, CL_DEVICE_MAX_COMPUTE_UNITS, *comp_unit);
+				snprintf(comp_unit_type, MAXSTR, "%s", "EU"); // Execution Unit
+				break;
+			case 0x10DE:
+				ret_domain_nv = CLINFO(ocl_dev_id, CL_DEVICE_PCI_DOMAIN_ID_NV, ocl_domain_nv);
+				ret_bus_nv    = CLINFO(ocl_dev_id, CL_DEVICE_PCI_BUS_ID_NV,    ocl_bus_nv);
+				ret_dev_nv    = CLINFO(ocl_dev_id, CL_DEVICE_PCI_SLOT_ID_NV,   ocl_dev_nv); // Slot == Device
+
+				if (ret_domain_nv && ret_bus_nv && ret_dev_nv)
+				{
+					MSG_WARNING(_("OpenCL Driver is not support CL_DEVICE_PCI_DOMAIN_ID_NV, CL_DEVICE_PCI_BUS_ID_NV, CL_DEVICE_PCI_SLOT_ID_NV"), NULL);
+					return ret_domain_nv && ret_bus_nv && ret_dev_nv;
+				}
+
+				if (dev->domain == ocl_domain_nv &&
+				    dev->bus    == ocl_bus_nv    &&
+				    dev->dev    == ocl_dev_nv)
+				{
+					CLINFO(ocl_dev_id, CL_DEVICE_MAX_COMPUTE_UNITS, *comp_unit);
+					snprintf(comp_unit_type, MAXSTR, "%s", "SM"); // Streaming Multiprocessor
+				}
+				break;
+			default:
+				MSG_DEBUG("get_gpu_comp_unit function support only AMD/Intel/NVIDIA GPU", NULL);
+				break;
+			} /* end switch (vendor_id) */
+		} /* end num_ocl_dev */
+	} /* end num_pf */
+#else
```

Comment:
Please add `UNUSED(dev);` also to avoid a warning when OpenCL support is disabled.